Ready for code review

noomdalv opened the pull request.

noomdalv committed:
added html structure and headers navbars positioning and css

SofiBretz committed:
1st commit - last header info. Main columns and rows in process.

noomdalv committed:
added 2nd to 4th sections with css grid positioning

noomdalv committed:
added section

SofiBretz committed:
2nd commit - first responsive prototype finished.

noomdalv committed:
Merge pull request #1 from noomdalv/feature-mainprototype

Feature mainprototype

SofiBretz committed:
3rd commit - resposive media in process

noomdalv committed:
added css positioning for first section

SofiBretz committed:
4th commit - responsive prototype ready.

noomdalv committed:
fixed column positioning for smallscreens

SofiBretz committed:
5th commit - logo first section

noomdalv committed:
added gradient for firstSectionLeft box

noomdalv committed:
added black gradient for container

SofiBretz committed:
6th commit - icons and titles to first section

noomdalv committed:
fixed positioning for firstSection grid containers

SofiBretz committed:
7th commit - fixing the grid first section in process

noomdalv committed:
added positioning for first grid right container elements

SofiBretz committed:
8th commit - firstSectionRight2 completed.

noomdalv committed:
added second section content and img positioning

SofiBretz committed:
9th commit - third section in process

noomdalv committed:
finished 2nd section

SofiBretz committed:
10th commit- third section items

noomdalv committed:
added css positioning for third section header containers

noomdalv committed:
finished thirdSectionMain and footer positioning css

SofiBretz committed:
11th commit - third section finished

noomdalv committed:
finished css positioning and styling for 3rd section footer

SofiBretz committed:
12th commit - fourth section in process

noomdalv committed:
added Latest container positioning for fourthSection

SofiBretz committed:
13th commit - fourth section list ready

noomdalv committed:
finished fifth section

SofiBretz committed:
14th commit - fourth section finished

noomdalv committed:
finished fourth section list styling and positioning

SofiBretz committed:
15th commit - footer box1 finished, box2 in process.

noomdalv committed:
finished main page css

noomdalv committed:
fixed container sizes for first section

SofiBretz committed:
16th commit - first section titles finished

noomdalv committed:
added media query for first section

SofiBretz committed:
17th commit - fixing the responsive first section

noomdalv committed:
finished 1st section mobile

SofiBretz committed:
18th commit - fixing the first section responsive part

noomdalv committed:
added positioning for 2nd section mobile version

SofiBretz committed:
19th commit - fixing second section media

noomdalv committed:
finished positioning for 2nd section mobile

SofiBretz committed:
20th commit - media fixed on section 3rd, 4th, 5th and footer.

noomdalv committed:
fixed image fifth section, ready for code review

noomdalv committed:
updated README.md

bolah2009 commented:
Currently under review

bolah2009 commented on css/main.css:
```diff
@@ -0,0 +1,866 @@
+
+  .firstGridLeft h2 {
+    display: flex;
+    flex-direction: wrap;
```
`SUGGESTION`

Consider changing to a correct value.

`flex-direction` does not have a `wrap` value. Possible values are `row` `row-reverse`, `column`, `column-reverse` and other global attributes. Learn more about `flex-direction` [here](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction)

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,743 @@
+            <a href="https://thenextweb.com/syndication/2019/07/29/hype-is-killing-ai-heres-how-can-we-can-stop-it/">Hype is killing AI – here’s how we can stop it</a>
+          </h2>
+          <ul>
+            <li><a id="firstSectionTitleA" href="https://thenextweb.com/author/ben-dickson/">by Ben Dickson</a></li>
```
`SUGGESTION`

Consider using a unique ID. [See here](https://github.com/noomdalv/Building-with-Responsive-Design/pull/2#discussion_r310845568)

bolah2009 commented on index.html:
```diff
@@ -0,0 +1,743 @@
+            <a href="https://thenextweb.com/syndication/2019/07/29/hype-is-killing-ai-heres-how-can-we-can-stop-it/">Google ad confirms the Pixel 4's face unlock and handwave-y gestures</a>
+          </h2>
+          <ul>
+            <li><a id="firstSectionTitleA" href="https://thenextweb.com/author/napierLopez/">by Napier Lopez</a></li>
```
`SUGGESTION`

You seem to be using this ID more than once, as it appears [here](https://github.com/noomdalv/Building-with-Responsive-Design/pull/2#discussion_r310845420)

bolah2009 reviewed (changes requested):
## Nice Job Team :clap: 

The clone looks nice, however, I would suggest making it more responsive since the main objective of the project is to make a responsive clone of [this page](https://thenextweb.com/). See below (check the full [page](https://thenextweb.com/) for more section that needs to be responsive:

| Current | Preferred |
| :-----: | :-----: |
| ![tnw current](https://user-images.githubusercontent.com/36057474/62503656-0e815580-b7ed-11e9-8cf5-f1636bf89d83.gif) | ![tnw preffered](https://user-images.githubusercontent.com/36057474/62503665-18a35400-b7ed-11e9-9de5-d00f7707be17.gif) |

You can recreate this issue (Image above) by using the console on chrome **(Inspect `Ctrl+Shift+I`)** and the **Toggle device toolbar** `Ctrl+Shift+M`. This way you can simulate how the app will render on small to widescreen devices.

### Optional but strongly recommended

- Note [this comment](#discussion_r310841076) on using `wrap` as `flex-direction` value.

- Check out [these errors](https://validator.w3.org/nu/?doc=https%3A%2F%2Fraw.githack.com%2Fnoomdalv%2FBuilding-with-Responsive-Design%2Ffeature-smallscreenprototype%2Findex.html) too.

Good luck with your next submission.

Cheers
[Bola](https://bolabuari.com/)

noomdalv committed:
fixed wc3 errors and media queries for 2nd / 3rd section mobile version

SofiBretz committed:
Merge branch 'feature-smallscreenprototype' of https://github.com/noomdalv/Building-with-Responsive-Design into feature-smallscreenprototype

noomdalv committed:
removed border

SofiBretz committed:
Update README.md

readme

SofiBretz committed:
Merge branch 'feature-smallscreenprototype' of https://github.com/noomdalv/Building-with-Responsive-Design into feature-smallscreenprototype

SofiBretz committed:
Update main.css

21th commit - 4th media section ready, 5th section media in process.

noomdalv committed:
mobile version fix

fixed fifth section positioning for mobile version

SofiBretz committed:
footer media

22th commit - fixing the footer media

noomdalv committed:
ready for 2nd review

fixed mobile positioning

noomdalv committed:
fixed navbar

navbar position for smallscreen

SofiBretz committed:
23th commit - header media fixed, logo media in process

noomdalv committed:
fixed navbar white

added navbar white items for mobile version

SofiBretz committed:
24th commit - header, 4th and 5th section media fixed.

noomdalv committed:
added display grid containers for <header> and <main>, fixed navbar

navbar white positioning for mobile

SofiBretz committed:
25th commit - footer media fixed. Ready to code review.

SofiBretz committed:
26th commit - W3 Validator recommendations fixed.

noomdalv committed:
added stickler

noomdalv committed:
stickler reviews fixed

noomdalv committed:
stickler test review

noomdalv committed:
stickler fixes

noomdalv committed:
ready for review

finally

Torres-ssf reviewed (changes requested):
### Nicely done team :clap: 
But before I give a :+1: we need to do a few changes to make your clone page behaves just like the original one (these are required by Microverse guidelines). They are:

- Latest news section

- [ ] Must aligns 3 articles per row in the medium version(4 is only for the large version).

- Latest Funding section

- [ ] Must aligns 3 articles per row in the medium version(4 is only for the large version).

- Categories (Apps, GERA, Tech, etc) section

- [ ] Must aligns 2 articles per row in the medium version(3 is only for the large version).

- Latest deals section

- [ ] Must aligns 3 articles per row in the medium version(4 is only for the large version).

Please kindly make these changes and submit for code review again!

noomdalv committed:
added 1024px breakpoint

all request changes are completed, ready for submission.

noomdalv committed:
Update README.md

mosaaleb reviewed (changes requested):
## Well done @noomdalv, @SofiBretz :clap:

I see that you have put a lot of effort into this project. I admire your dedication but there is some requirement needs to be met before moving on.

### Large Screen:
- [ ] Navigation bar contain search icon at the rightmost side -- As seen on the the original page.


### Medium Screen

- [ ] Links are hiding behind the blocks as you can see. let's fix this
- [ ] Also each funding section header text is hiding behind the left thumbnail in All screen sizes.
![Screenshot from 2019-08-08 19-57-02](https://user-images.githubusercontent.com/4246637/62726134-c9d90280-ba16-11e9-9680-3d8e3670cbdd.png)

- [ ] Please, use [stickler](https://github.com/microverseinc/linters-config/tree/master/css). It doesn't take time to use and it makes sure that you have good styling guidelines.

- [ ] [OPTIONAL]: I highly recommend the start all of the coming projects using [`reset.css`] (https://meyerweb.com/eric/tools/css/reset/) file (except when you are using bootstrap as it is bundled by default with a reset file) , as you can you see there is some margin at the top of the page, this is default value and can change by browsers. Using `reset.css` file ensure that we have the same initial state in all browsers.


Please handle these requirements and submit a new code review.

### Happy Coding :computer:

SofiBretz committed:
27th commit - search icon on big screen

noomdalv committed:
final code review, fixed all errors

please review

noomdalv committed:
Update README.md

bolah2009 commented:
Currently under review

bolah2009 commented on css/main.css:
```diff
@@ -0,0 +1,991 @@
+}
+
+@media screen and (max-width: 768px) {
+  #fifthSection {
```
`HINT`

You seem to be using only two breakpoints for the media query which is affecting the display of this section for smaller screens. See below:

| Current | Preferred |
| :-----: | :-----: |
| ![tnw current](https://user-images.githubusercontent.com/36057474/62747638-47b80080-ba4d-11e9-96b5-5efafb919936.png) | ![tnw preffered](https://user-images.githubusercontent.com/36057474/62747646-4d154b00-ba4d-11e9-86df-c51d342cb631.png) |

Ispirett commented:
# Reviewing

bolah2009 reviewed (approved):
## Good Job Team :clap: 

The [clone](https://raw.githack.com/noomdalv/Building-with-Responsive-Design/feature-smallscreenprototype/index.html) looks good to me.

#### Friendly reminder: You should use [Linter](https://github.com/microverseinc/linters-config#linters-config) in this project. Please make sure you use it in the next project. You will find instructions in the Pathwright lesson description.

### Check the comment below before merging.

Good luck with your next project.

Cheers
[Bola](https://bolabuari.com/)

noomdalv committed:
Ready

fixed Latest Deals section for small screens, it should display a row for every element now

noomdalv commented:
Hi Bola, thank you for the review!
i just fixed the error and it should be ok now, last commit we were
changing something for the 1024px breakpoint and somehow missed that,
thanks again.
the latest deals section displays the elements in rows now, like this:

[image: image.png]


Bye!.

El jue., 8 de ago. de 2019 a la(s) 20:39, BOLA AHMED BUARI (
notifications@github.com) escribió:

> *@bolah2009* approved this pull request.
> Good Job Team 👏
>
> The clone
> <https://raw.githack.com/noomdalv/Building-with-Responsive-Design/feature-smallscreenprototype/index.html>
> looks good to me.
> Friendly reminder: You should use Linter
> <https://github.com/microverseinc/linters-config#linters-config> in this
> project. Please make sure you use it in the next project. You will find
> instructions in the Patwright lesson description. Check the comment below
> before merging.
>
> Good luck with your next project.
>
> Cheers
> Bola <https://bolabuari.com/>
> ------------------------------
>
> In css/main.css
> <https://github.com/noomdalv/Building-with-Responsive-Design/pull/2#discussion_r312301865>
> :
>
> > +}
>
> +
>
> +@media screen and (max-width: 1024px) {
>
> +  #fifthSection {
>
> +    grid-template-columns: repeat(3, 1fr);
>
> +    grid-template-rows: 40px 320px 320px 320px;
>
> +    margin-bottom: -50px;
>
> +  }
>
> +
>
> +  #latestDeals {
>
> +    grid-column: 1 / 4;
>
> +  }
>
> +}
>
> +
>
> +@media screen and (max-width: 768px) {
>
> +  #fifthSection {
>
>
> HINT
>
> You seem to be using only two breakpoints for the media query which is
> affecting the display of this section for smaller screens. See below:
> Current Preferred
> [image: tnw current]
> <https://user-images.githubusercontent.com/36057474/62747638-47b80080-ba4d-11e9-96b5-5efafb919936.png> [image:
> tnw preffered]
> <https://user-images.githubusercontent.com/36057474/62747646-4d154b00-ba4d-11e9-86df-c51d342cb631.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/noomdalv/Building-with-Responsive-Design/pull/2?email_source=notifications&email_token=AMKWIWSRDO2DXDPEVY5VBL3QDTDEFA5CNFSM4IJPOR62YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCBCELSY#pullrequestreview-272909771>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMKWIWVOXIM6J2TMPINCXWTQDTDEFANCNFSM4IJPOR6Q>
> .
>

noomdalv committed:
Update README.md
